### PR TITLE
Closes #243: Refactor train test splits

### LINKED ISSUE #243: Refactor the training stability evaluations
Refactor the existing evaluation for [training stability](https://github.com/mozilla/PRESC/tree/master/presc/training_stability) to conform with the new design for `presc.evaluations`

### PR DESCRIPTION
**Pull request by dzeber**
[Merge #269 first]

Porting the train-test-splits analysis to the current design.

Leaving tests for later given time constraints.

### CODE REVIEW

**Comment by dzeber** [comment]
@gleonard-m Take a quick look only if you have time given short notice. I haven't added test cases yet for the new code, but it runs as a part of the report.

This branch continues from the branch from #269 (to minimize merge conflicts) so that should be merged first.

File: presc/report/resources/_config.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ only_build_toc_files: true
 # Force re-execution of notebooks on each build.
 execute:
   execute_notebooks: force
+  timeout: 300
```
Comment (gleonard-m) [inline presc/report/resources/_config.yml:9]: You noticed a timeout issue when running a report of this size?

File: presc/report/resources/landing.ipynb [hunk 2]
```diff
@@ -89,8 +89,8 @@
    "source": [
     "#print(f\"Training set size: {train_dataset.size:,}\")\n",
     "print(\n",
-    "    f\"Test set size: {test_dataset.size:,}\" +\n",
-    "    f\" ({test_dataset.size / dataset.size:.1%})\"\n",
+    "    f\"Test set size: {test_dataset.size:,}\"#+\n",
+    "#    f\" ({test_dataset.size / dataset.size:.1%})\"\n",
     ")"
```
Comment (gleonard-m) [inline presc/report/resources/landing.ipynb:94]: nit - remove commented out code

**Review by gleonard-m — APPROVED** [review]

File: presc/report/resources/_config.yml (hunk 1 above)
Comment (dzeber) [reply presc/report/resources/_config.yml:9]: The default timeout was 30 seconds, and that tripped on one of the test cases. I thought bump this up to give some more leeway.

File: presc/report/resources/landing.ipynb (hunk 2 above)
Comment (dzeber) [reply presc/report/resources/landing.ipynb]: Done

**Pull request merged**